src/components/Sidebar/Cart.js

```diff
@@ -2,21 +2,63 @@ import React from 'react';
 import Typography from '@material-ui/core/Typography';
 import { withStyles } from '@material-ui/core/styles';
 import PropTypes from 'prop-types';
+import { UnstyledSection } from '../Calendar/Section';
 
 export const styles = {
   cartHeading: {
     padding: '16px',
   },
 };
 
+export const sectionStyles = section => (
+  {
+    paper: {
+      position: 'static',
+      backgroundColor: section.color,
+      overflow: 'hidden',
+    },
+    container: {
+      margin: '3px',
+    },
+    text: {
+      color: 'white',
+    },
+    header: {
+      display: 'flex',
+      justifyContent: 'space-between',
+      flexWrap: 'wrap',
+    },
+  }
+);
+
 function Cart({ classes, sections }) {
-  const label = sections.length === 1 ? 'Class' : 'Classes';
+  const uniqueSections = [];
+  const usedIDs = [];
+
+  sections.forEach(
+    (section) => {
+      if (!usedIDs.includes(section.id)) {
+        usedIDs.push(section.id);
+        uniqueSections.push(section);
+      }
+    },
+  );
+
+  const label = uniqueSections.length === 1 ? 'Class' : 'Classes';
+
   return (
-    <Typography variant="h5" className={classes.cartHeading}>
-      {sections.length}
-      {' '}
-      {label}
-    </Typography>
+    <div>
+      <Typography variant="h5" className={classes.cartHeading}>
+        {uniqueSections.length}
+        {' '}
+        {label}
+      </Typography>
+      {uniqueSections.map((section) => {
```

Comment:
This is a great teaching moment for idiomatic React! If you're having to wrap something in render with a `withStyles` HOC, this should act as a warning sign to the fact that you're essentially creating a new component. This new component should be extracted and encapsulated in its own file, then imported and used here. This compositional pattern of React is what makes it so great to write & powerful!

Reply:
@mfdong ^